docs/gettingstarted/mint_tokens.md

```diff
@@ -16,6 +16,118 @@ nav_order: 9
+- Each pool is classified as _fungible_ or _non-fungible_
+- In the case of _non-fungible_ tokens, the pool is subdivided into individual tokens with a unique _token index_
+- Within a pool, you may _mint_, _transfer_, and _burn_ tokens
+- Each operation can be optionally accompanied by a broadcast or private message, which will be recorded alongside the transfer on-chain
+- FireFly tracks a history of all token operations along with all current token balances
+- The blockchain backing each token connector may be the same _or_ different from the one backing FireFly message pinning
+
+## Create a pool
+
+Every application will need to create at least one token pool. At a minimum, you must always
+specify a `name` and `type` (fungible or nonfungible) for the pool.
+
+`POST` `/api/v1/namespaces/default/tokens/pools`
+
+```json
+{
+  "name": "testpool",
+  "type": "fungible"
+}
+```
+
+Other parameters:
+- You must specify a `connector` if you have configured multiple token connectors
+- You may pass through a `config` object of additional parameters, if supported by your token connector
+- You may specify a `key` understood by the connector (i.e. an Ethereum address) if you'd like to use a non-default signing identity
+
+## Mint tokens
+
+Once you have a token pool, you can mint tokens within it. With the default `firefly-tokens-erc1155` connector,
+only the creator of a pool is allowed to mint - but each connector may define its own permission model.
+
+`POST` `/api/v1/namespaces/default/tokens/mint`
+
+```json
+{
+  "amount": 10
+}
+```
+
+Other parameters:
+- You must specify a `pool` name if you've created more than one pool
+- You may specify a `key` understood by the connector (i.e. an Ethereum address) if you'd like to use a non-default signing identity
+- You may specify `to` if you'd like to send the minted tokens to a specific identity (default is the same as `key`)
+
+## Transfer tokens
+
+You may transfer tokens within a pool by specifying an amount and a destination understood by the connector (i.e. an Ethereum address).
+With the default `firefly-tokens-erc1155` connector, only the owner of a token may transfer it away - but each connector may define its
+own permission model.
+
+`POST` `/api/v1/namespaces/default/tokens/transfers`
+
+```json
+{
+  "amount": 1,
+  "to": "0x07eab7731db665caf02bc92c286f51dea81f923f"
+}
+```
+
+Other parameters:
+- You must specify a `pool` name if you've created more than one pool
+- You must specify a `tokenIndex` for non-fungible pools (and the `amount` should be 1)
+- You may specify a `key` understood by the connector (i.e. an Ethereum address) if you'd like to use a non-default signing identity
+- You may specify `from` if you'd like to send tokens from a specific identity (default is the same as `key`)
+
+## Sending data with a transfer
+
+All transfers (as well as mint/burn operations) support an optional `message` parameter that contains a broadcast or private
```

Comment:
I think we effectively hijacked the `_data` field of the various operations that may have been needed for further on-chain authorizations (e.g signatures by parties other than the sender to indicate approval). as a mental note we should consider supporting an additional payload parameter to include into the `_data` field as-is

Reply:
@jimthematrix yes, you're correct. The token connector exposes a `data` field, and FireFly packs JSON into this field to identify messages and other metadata. But FireFly itself does not (currently) expose any way for the user to control or add to this field from the level of the FireFly APIs.